### DISCUSSION TITLE
How to use time 't' in a current function?

### DISCUSSION THREAD

**Discussion by cyasing** [pybamm-team/PyBaMM#2875] [Q&A] [answered]
I tried  programming a current function like this:
```
def controlled_current(one_c, c_rate):
    def current(t):
        return float(one_c if t <= 1000 else (-1*one_c*c_rate[int(t-1001)]))    
    return current
```
where the c_rate will be an array of c-rates over a hundred seconds or so. I passed it as an input to the solver via current function, but it doesn't work due to not being able to use time "t" as a number (All it returns is a Time object). What should I do? The way I input is through the `param["Current function [A]"] = controlled_current(one_c, c_rate)`.

**Comment by cyasing** [comment]
Edited the question based on changes I made that worked for simpler trial of function.

**Reply by RuiheLi** [reply]
Hi~ Yes "t" is by default a variable in PyBaMM. I think the best reference is [this notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/Getting%20Started/Tutorial%204%20-%20Setting%20parameter%20values.ipynb), where you can find a drive cycle section and custom current section. 
Also, it may be very helpful if you can kindly post your full scripts here, or explain a bit more how you are trying to control the current? Thanks!

**Comment by cyasing** [comment]
Turns out that the only way to access time dependent current is through step simulation.

**Reply by cyasing** [reply]
In case a function requires to make time-based decisions.

**Reply by rtimms** [reply]
You can use `<` and `>` to define a heaviside function. E.g.
```
def current(t):
    return t < 100
```
would return a current of 1 when t is less than 100 seconds and 0 otherwise. You can combine as many as you like to make the function you want.

**Reply by rtimms** [reply, marked as answer]
Alternatively you could make an experiment. If you have an array of rates you could make the experiment string programmatically, e.g. construct it in a loop.

**Reply by cyasing** [reply]
I did go for the second option in the end. I'll see if the simpler expression works.
